Apps/WebClient/src/ClientApp/src/components/modal/covid.vue

```diff
@@ -24,104 +22,102 @@ export default class CovidModalComponent extends Vue {
         userPreference: UserPreference;
     }) => void;
     @Getter("user", { namespace: "user" }) user!: User;
+    @Getter("laboratoryOrders", { namespace: "laboratory" })
+    laboratoryOrders!: LaboratoryOrder[];
 
-    @Prop() error!: boolean;
-    @Prop({ default: false }) isLoading!: boolean;
+    @Getter("getPreference", { namespace: "user" })
+    getPreference!: (preferenceName: string) => UserPreference | undefined;
 
-    private logger!: ILogger;
+    @Prop({ default: false }) isLoading!: boolean;
 
-    private mounted() {
-        this.logger = container.get<ILogger>(SERVICE_IDENTIFIER.Logger);
+    private isDismissed = false;
+
+    private get isVisible(): boolean {
```

Comment:
The logic for showing this modal now resides on it instead of the Timeline. That way there is no need to sync the events on the timeline with the state required for showing the covid alert modal